### PR TITLE
PaidLists now store if the CostPart source is linked to host

### PR DESCRIPTION
**Pull request by tool4ever**
Part for #2244:
Casting _Corbse Cobble_ via _Maestros Ascendancy_ now correctly only uses the stats from the original cost for the token P/T.

Some bonus cleanup especially in _CardUtil_, the removed parts turned redundant to ``SpellAbility.canTarget`` checks.

### CODE REVIEW

File: forge-game/src/main/java/forge/game/card/CardUtil.java [hunk 1]
```diff
@@ -550,66 +549,21 @@ public static List<Card> getValidCardsToTarget(TargetRestrictions tgt, SpellAbil
...
                 choices.remove(activatingCard);
             }
         }
-        List<GameObject> targetedObjects = ability.getUniqueTargets();
 
         // Remove cards already targeted
         final List<Card> targeted = Lists.newArrayList(ability.getTargets().getTargetCards());
         choices.removeAll(targeted);
 
-        // Remove cards exceeding total CMC
```
Comment (Hanmac) [inline forge-game/src/main/java/forge/game/card/CardUtil.java:569]: why is that removed? i think that needs to be checked there too

**Comment by Hanmac** [comment]
cost parts that was marked intrinsic and costs that aren't shouldn't be merged?

**Comment by tool4ever** [comment]
There is one exception by the rules which is for _Carth the Lion_

but yes other than that ``CostAdjustment`` uses ``Cost.mergeTo()`` which I already implemented in #2253
It should make sure that all different CostParts are kept apart

(Unfortunately this temporarily slighty increased cheating potential via GUI canceling a partial payment after costs that can't refund but that needs to be fixed better in a different place anyway)

**Comment by Hanmac** [comment]
For example: `Rescue from the Underworld`
> As an additional cost to cast this spell, sacrifice a creature.

i want this part to be a Static Ability on the card instead of `Cost$` 

then `Drought`
> Spells cost an additional “Sacrifice a Swamp” to cast for each black mana symbol in their mana costs.

so BOTH of them should be done as Static Abilities, but only the first one should be treated as intrinsic, the second should not.

so i mean this costs should not be merged (even if the valid part would be the same)

**Comment by tool4ever** [comment]
For the refactoring we have this: #2472

But I'm not sure it's high priority because even without it's already working now

Part 1 (#2253) stopped the merging so they are paid as different cost parts like it should be
Part 2 (this) turns the paidLists into Table so linked abilities can filter for the right ones

**Review by kevlahnota — APPROVED** [review]
